### PR TITLE
BaseRecord: allow find_by() and where() with any attribute

### PR DESCRIPTION
**Pull request by Fabien-Wecasa**
The goal here is to use `find_by` and `where` with any attribute. E.g. to be able to write `Zoho::Lead.find_by(last_name: "Snow")`

### CODE REVIEW

File: lib/zoho_hub/base_record.rb [hunk 1]
```diff
@@ -42,6 +42,17 @@ def find(id)
       def where(params)
         path = File.join(request_path, 'search')
 
+        if params.one?
```
Comment (rikas) [inline lib/zoho_hub/base_record.rb]: I would prefer to not use a Rails only method here.

**Comment by rikas** [comment]
@Fabien-Wecasa I'm out now, so I can't review this properly, but I added a comment in your code, if you could already fix that I would appreciate it.

Thanks!

File: lib/zoho_hub/base_record.rb (hunk 1 above)
Comment (Fabien-Wecasa) [reply lib/zoho_hub/base_record.rb]: Good point !
:white_check_mark:

**Comment by Fabien-Wecasa** [comment]
@rikas Maybe some explanations could help :smiley: 
Zoho natively handles searches by [some criteria](https://www.zoho.com/crm/help/developer/api/search-records.html), like `email` or `phone`, or a more generic search by `criteria`. I need to search by other attributes, e.g. `like_name`. I want to write `find_by(last_name: 'XXX')` instead of writing `find_by(criteria: "Last_Name:equals:XXX")`.
For this, I need to use `attr_to_zoho_key` as a class method, hence the changes in `with_attributes.rb`.

The drawback of this approach is to hide the complexity of a search in Zoho: a search by email or phone is faster than a search by another attribute. But I find it useful anyway...

**Comment by rikas** [comment]
Looks good. Thanks 👍

**Pull request merged**